pkg/kernel/kernel.go

```diff
@@ -0,0 +1,52 @@
+package kernel
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/alibaba/pouch/pkg/exec"
+	"github.com/pkg/errors"
+)
+
+// VersionInfo holds information about the kernel.
+type VersionInfo struct {
+	Kernel int    // Version of the kernel (e.g. 4.1.2-generic -> 4)
+	Major  int    // Major part of the kernel version (e.g. 4.1.2-generic -> 1)
+	Minor  int    // Minor part of the kernel version (e.g. 4.1.2-generic -> 2)
+	Flavor string // Flavor of the kernel version (e.g. 4.1.2-generic -> generic)
+}
+
+// String returns the kernel version's string format.
+func (k *VersionInfo) String() string {
+	return fmt.Sprintf("%d.%d.%d-%s", k.Kernel, k.Major, k.Minor, k.Flavor)
+}
+
+// GetKernelVersion returns the kernel version info.
+func GetKernelVersion() (*VersionInfo, error) {
+	var (
+		kernel, major, minor int
+		flavor               string
+	)
+
+	_, stdout, _, err := exec.Run(0, "uname", "-a")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to run command uname -a")
+	}
+
+	arr := strings.Split(stdout, " ")
+	if len(arr) < 3 {
+		return nil, fmt.Errorf("uname format [%s] is invalid", stdout)
+	}
+
+	parsed, _ := fmt.Sscanf(arr[2], "%d.%d.%d-%s", &kernel, &major, &minor, &flavor)
```

Comment:
If use uname -r, you may don't need to check slip the output, as it only has the version info .

Reply:
yes